### ISSUE TITLE
Proper suport for scales in 2D and 3D layers

### ISSUE THREAD

**Issue by jni** [napari/napari#436]
## 🚀 Feature

Currently, scale on a layer can be set with `layer.scale = <tuple>`, which is great. However, this method currently sets vispy scale directly (?) and so does not follow row/column aka zyx coordinates as we do. For example, in the following code:

```python
from skimage import data
import napari
v = napari.Viewer()
image_layer = v.add_image(data.camera())
image_layer.scale = (1, 2)
```

The image is stretched along the *vertical* dimension, when NumPy axis conventions would lead us to expect that the horizontal dimension is stretched.

The scale parameter should be set with a property + setter to properly transpose the coordinate scales.

**Comment by pranathivemuri** [comment]
we can allow it to be a parameter that can be tuned on the layer. I can add that it in. But I was wondering if we could set the whole 3  tuple of scale at once or do we want slider buttons for each of the scales in x y and z. We transpose/reverse the scales set on the layer from xyz to zyx and xy to yx in the property setter

**Comment by jni** [comment]
@pranathivemuri imho we don't want a slider: the scale is something that is fixed for each image. Eventually we'll want to adjust it based on metadata, but an input parameter works great. I agree about transposing the scales in the property setter. =) Thanks!

**Comment by constantinpape** [comment]
I agree with @jni that this should be a fixed parameter (and that it should respect the same axis-convention as napari). IMHO, it should be a keyword argument for `add_image` and other relevant functions.

Also, is this supported for the pyramid layer already? I.e. the scale of each level should be something like `scale_of_level = [sc1 * sc2 for sc1, sc2 in zip(scale, downsample_level)]`, where `scale` is the scale factor of level 0 and `downsample_level` the downsampling factor of the current level.

**Comment by sofroniewn** [comment]
This has now been mostly implemented in #493 - each layer accepts a `scale` keyword arg that is in the axis conventions of napari.

inside the pyramid layer the `scale` right now gets autocomputed based on the downsampling, but the proposal by @constantinpape to have both a `scale` and `scale_of_level` would be preferable. I'll implement that shortly

Similarly the `translate` property is now a keyword arg on every layer, but in pyramids is getting overloaded and an additional `translate_of_view` property would be much cleaner.
